smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/serde/member/MemberShapeDecodeGenerator.kt

```diff
@@ -147,6 +156,40 @@ open class MemberShapeDecodeGenerator(
     private fun default(memberShape: MemberShape): String {
         val targetShape = ctx.model.expectShape(memberShape.target)
         val defaultTrait = memberShape.getTrait<DefaultTrait>() ?: targetShape.getTrait<DefaultTrait>()
+        val requiredTrait = memberShape.getTrait<RequiredTrait>()
+        // If member is required but there isn't a default value, use zero-equivalents for error correction
+        if (requiredTrait != null && defaultTrait == null) {
+            return when (targetShape) {
+                is EnumShape, is IntEnumShape -> " ?? .sdkUnknown(\"\")"
+                is StringShape -> " ?? \"\""
+                is ByteShape, is ShortShape, is IntegerShape, is LongShape -> " ?? 0"
+                is FloatShape, is DoubleShape -> " ?? 0.0"
+                is BooleanShape -> " ?? false"
+                is ListShape -> " ?? []"
+                is MapShape -> " ?? [:]"
+                is TimestampShape -> " ?? Date(timeIntervalSince1970: 0)"
+                is DocumentShape -> {
+                    val underlyingNode = requiredTrait.toNode()
+                    when (underlyingNode.type) {
+                        NodeType.OBJECT -> { " ?? Document.object([:])" }
+                        NodeType.ARRAY -> { " ?? Document.array([])" }
+                        NodeType.BOOLEAN -> { " ?? Document.boolean(false)" }
+                        NodeType.STRING -> { " ?? Document.string(\"\")" }
+                        NodeType.NUMBER -> { " ?? Document.number(0)" }
+                        NodeType.NULL -> { throw CodegenException("Unreachable statement") } // This will never happen
+                    }
+                }
+                is BlobShape -> {
+                    if (targetShape.hasTrait<StreamingTrait>()) {
+                        " ?? ByteStream.data(\"\".data(using: .utf8))"
+                    } else {
+                        " ?? \"\".data(using: .utf8)"
+                    }
+                }
+                // No default provided for other types
+                else -> { "" }
+            }
+        }
```

Comment:
Change here adds server-side serialization error where it fails to send values for required members of the output. In that case, the SDK fills it with zeros or zero-equivalents.